app/assets/javascripts/discourse/tests/acceptance/user-menu-test.js

```diff
@@ -178,6 +186,190 @@ acceptance("User menu", function (needs) {
+      "has a link to the preferences page of the user"
+    );
+    assert.strictEqual(
+      preferencesLink.textContent.trim(),
+      I18n.t("user.preferences"),
+      "preferences link has the right label"
+    );
+    assert.ok(
+      preferencesLink.querySelector(".d-icon-cog"),
+      "preferences link has the right icon"
+    );
+
+    let doNotDisturbButton = query(
+      "#quick-access-profile ul li.do-not-disturb .btn"
+    );
+    assert.strictEqual(
+      doNotDisturbButton.textContent
+        .replaceAll(/\s+/g, " ")
+        .replaceAll(/\u200B/g, "")
+        .trim(),
+      I18n.t("do_not_disturb.label"),
+      "Do Not Disturb button has the right label"
+    );
+    assert.ok(
+      doNotDisturbButton.querySelector(".d-icon-toggle-off"),
+      "Do Not Disturb button has the right icon"
+    );
+
+    await click("header.d-header"); // close the menu
+    const date = new Date();
+    date.setHours(date.getHours() + 2);
+    updateCurrentUser({ do_not_disturb_until: date.toISOString() });
+    await click(".d-header-icons .current-user");
+    await click("#user-menu-button-profile");
+
+    doNotDisturbButton = query(
+      "#quick-access-profile ul li.do-not-disturb .btn"
+    );
+    assert.strictEqual(
+      doNotDisturbButton.textContent
+        .replaceAll(/\s+/g, " ")
+        .replaceAll(/\u200B/g, "")
+        .trim(),
+      `${I18n.t("do_not_disturb.label")} 2h`,
+      "Do Not Disturb button has the right label when Do Not Disturb is enabled"
+    );
+    assert.ok(
+      doNotDisturbButton.querySelector(".d-icon-toggle-on"),
+      "Do Not Disturb button has the right icon when Do Not Disturb is enabled"
+    );
+
+    let toggleAnonButton = query(
+      "#quick-access-profile ul li.enable-anonymous"
+    );
+    assert.strictEqual(
+      toggleAnonButton.textContent
+        .replaceAll(/\s+/g, " ")
+        .replaceAll(/\u200B/g, "")
+        .trim(),
+      I18n.t("switch_to_anon"),
+      "toggle anonymous button has the right label when the user isn't anonymous"
+    );
+    assert.ok(
+      toggleAnonButton.querySelector(".d-icon-user-secret"),
+      "toggle anonymous button has the right icon when the user isn't anonymous"
+    );
+
+    await click("header.d-header"); // close the menu
+    updateCurrentUser({ is_anonymous: true });
+    await click(".d-header-icons .current-user");
+    await click("#user-menu-button-profile");
+
+    toggleAnonButton = query("#quick-access-profile ul li.disable-anonymous");
+    assert.strictEqual(
+      toggleAnonButton.textContent
+        .replaceAll(/\s+/g, " ")
+        .replaceAll(/\u200B/g, "")
+        .trim(),
+      I18n.t("switch_from_anon"),
+      "toggle anonymous button has the right label when the user is anonymous"
+    );
+    assert.ok(
+      toggleAnonButton.querySelector(".d-icon-ban"),
+      "toggle anonymous button has the right icon when the user is anonymous"
+    );
```

Comment:
I would add a test case for the various conditions on when this link should show.

Reply:
O it has been added below :)